DynamicTablesPkg/Library/Acpi/Common/AcpiSlitLib/SlitGenerator.c

```diff
@@ -0,0 +1,412 @@
+    return EFI_INVALID_PARAMETER;
+  }
+
+  *Table = NULL;
+
+  SlitInfo                 = NULL;
+  DomainRelInfoCount       = 0;
+  NumberOfSystemLocalities = 0;
+  Status                   = GetEArchCommonObjSystemLocalityInfo (
+                               CfgMgrProtocol,
+                               CM_NULL_TOKEN,
+                               &SlitInfo,
+                               NULL
+                               );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SLIT: Failed to retrieve SLIT information. Status = %r\n",
+      Status
+      ));
+    return Status;
+  }
+
+  if (SlitInfo == NULL) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SLIT: No SLIT information provided by configuration manager.\n"
+      ));
+    return EFI_NOT_FOUND;
+  }
+
+  Status = GetEArchCommonObjProxDomainRelationInfo (
+             CfgMgrProtocol,
+             SlitInfo->RelativeDistanceArray,
+             &DomainRelationInfo,
+             &DomainRelInfoCount
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SLIT: Failed to retrieve Proximity Domain relation information. Status = %r\n",
+      Status
+      ));
+    return Status;
+  }
+
+  if ((DomainRelationInfo == NULL) || (DomainRelInfoCount == 0)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SLIT: No Proximity Domain relation information provided by configuration manager.\n"
+      ));
+    return EFI_NOT_FOUND;
+  }
+
+  for (Index = 0; Index < DomainRelInfoCount; Index++) {
+    if (DomainRelationInfo[Index].Relation < 0x10) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "ERROR: SLIT: Invalid relation value %d\n",
+        DomainRelationInfo[Index].Relation
+        ));
+      return EFI_INVALID_PARAMETER;
+    }
+
+    Status = GetEArchCommonObjProxDomainInfo (
+               CfgMgrProtocol,
+               DomainRelationInfo[Index].FirstDomainToken,
+               &DomainInfoFirst,
+               NULL
+               );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "ERROR: SLIT: Failed to retrieve First Proximity Domain information. Status = %r\n",
+        Status
+        ));
+      return Status;
+    }
+
+    Status = GetEArchCommonObjProxDomainInfo (
+               CfgMgrProtocol,
+               DomainRelationInfo[Index].SecondDomainToken,
+               &DomainInfoSecond,
+               NULL
+               );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "ERROR: SLIT: Failed to retrieve Second Proximity Domain information. Status = %r\n",
+        Status
+        ));
+      return Status;
+    }
+
+    if ((DomainInfoFirst == NULL) || (DomainInfoSecond == NULL)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "ERROR: SLIT: No Proximity Domain information provided by configuration manager.\n"
+        ));
+      return EFI_INVALID_PARAMETER;
+    }
+
+    if (DomainInfoFirst->DomainId == DomainInfoSecond->DomainId) {

```

Comment:
Also, there used to be a check where when `(FirstDomain != SecondDomain)`, the realtion value must be `< MAX_UINT8`